### DISCUSSION TITLE
dnsleak/whoer网站测试仍有国内运营商

### DISCUSSION THREAD

**Discussion by jasonding121205** [IrineSistiana/mosdns#296] [General]
插件工作正常，缓存提速效果明显。passwall上述网站已加入代理名单，但中国列表模式以外模式测试时发现并不能像使用v2rayn之类的客户端一样全部为国外dns。请问还需要如何设置吗？

**Comment by lbhzy** [comment]
> 插件工作正常，缓存提速效果明显。passwall上述网站已加入代理名单，但中国列表模式以外模式测试时发现并不能像使用v2rayn之类的客户端一样全部为国外dns。请问还需要如何设置吗？

dnsleak官网介绍：DNS 泄漏测试通过向您的客户端 **发送一系列域名** 以在特定测试域中解析来工作。
除非你把这一系列域名全部添加进黑名单，才可能全部为国外dns，我是这样理解的

**Reply by jasonding121205** [reply]
不知道我的理解对不对，passwall已经是中国列表以外模式了，发送的一系列域名在日志里观察也都是基于dnsleaktest.com的三级国外域名，按理应当是用forward remote里定义的dns解析

**Reply by IrineSistiana** [reply]
passwall 模式跟 mosdns 没关系。mosdns 有自己的域名分流策略。

如果你是用的其他软件集成了mosdns，你需要在那里反馈问题。

**Reply by jasonding121205** [reply]
感谢大佬亲自回复。fakedns能解决问题，但走的就不是mosdns监听端口了，还能用上mosdns的缓存吗？
